test/operation-hooks.suite/dao-patch-by-id.suite.js

```diff
@@ -0,0 +1,227 @@
+    });
+
+    it('triggers hooks in the correct order', function(done) {
+      TestModel.patchById(instanceId,
+        { name: 'changed' },
+        function(err, info) {
+          if (err) return done(err);
+          hookMonitor.names.should.eql([
+            'TestModel:before save',
+            'TestModel:persist',
+            'TestModel:loaded',
+            'TestModel:after save',
+          ]);
+          done();
+        });
+    });
+
+    it('triggers `before save` hook', function(done) {
+      TestModel.observe('before save', ctxRecorder.recordAndNext());
+
+      TestModel.patchById(instanceId, { name: 'changed' }, function(err, info) {
+        if (err) return done(err);
+        ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+          where: { id: instanceId },
+          data: { name: 'changed' },
+        }));
+        done();
+      });
+    });
+
+    it('aborts when `before save` hook fails', function(done) {
+      TestModel.observe('before save', nextWithError(expectedError));
+
+      TestModel.patchById(instanceId, { name: 'changed' }, function(err, info) {
+        [err].should.eql([expectedError]);
+        done();
+      });
+    });
+
+    it('applies updates from `before save` hook', function(done) {
+      TestModel.observe('before save', function(ctx, next) {
+        ctx.data.extra = 'extra data';
+        ctx.data.name = 'hooked name';
+        next();
+      });
+
+      TestModel.patchById(instanceId, { name: 'changed' }, function(err, info) {
+        if (err) return done(err);
+        TestModel.findById(instanceId, function(err, instance) {
+          if (err) return done(err);
+          should.exists(instance);
+
+          instance.toObject(true).should.eql({
+            id: instanceId,
+            name: 'hooked name',
+            extra: 'extra data',
+          });
+
+          done();
+        });
+      });
+    });
+
+    it('validates model after `before save` hook', function(done) {
+      TestModel.observe('before save', invalidateData);
+
+      TestModel.patchById(instanceId, { name: 'updated' }, function(err) {
+        err.should.be.instanceOf(ValidationError);
+        (err.details.codes || {}).should.eql({ name: ['presence'] });
+        done();
+      });
+    });
+
+    it('triggers `persist` hook', function(done) {
+      TestModel.observe('persist', ctxRecorder.recordAndNext());
+      TestModel.patchById(instanceId, { name: 'changed' }, function(err, info) {
+        if (err) return done(err);
+
+        ctxRecorder.records.should.eql(aCtxForModel(TestModel, {
+          where: { id: instanceId },
+          data: { name: 'changed' },
+          isNewInstance: false,
+        }));
+
+        done();
+      });
+    });
+
+    it('applies updates from `persist` hook', function(done) {
+      TestModel.observe('persist', ctxRecorder.recordAndNext(function(ctx) {
+        ctx.data.extra = 'hook data';
+      }));
+
+      TestModel.settings.updateOnLoad = true;
```

Comment:
This flag was added to preserve backwards compatibility, please don't use it in new features. Besides that, I believe `patchById` does not return full instance, thus this is not relevant at all.